test/integration-tests/IntegrationTests.StartupHook/StartupHookTests.cs

```diff
@@ -0,0 +1,216 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using IntegrationTests.Helpers;
+using IntegrationTests.Helpers.Mocks;
+using IntegrationTests.Helpers.Models;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IntegrationTests.StartupHook
+{
+    public class StartupHookTests : TestHelper
+    {
+        private List<WebServerSpanExpectation> _expectations = new List<WebServerSpanExpectation>();
+
+        public StartupHookTests(ITestOutputHelper output)
+            : base("StartupHook", output)
+        {
+            SetServiceVersion("1.0.0");
+            _expectations.Add(new WebServerSpanExpectation("Samples.StartupHook", null, "/alive-check", "/alive-check", null, "GET"));
+            _expectations.Add(new WebServerSpanExpectation("Samples.StartupHook", null, "/home/index", "/home/index", null, "GET"));
+            _expectations.Add(new WebServerSpanExpectation("Samples.StartupHook", null, "HTTP GET", "HTTP GET", null, "GET"));
+        }
+
+        [Theory]
+        [Trait("Category", "EndToEnd")]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void SubmitsTraces(bool enableCallTarget)
+        {
+            SetCallTargetSettings(enableCallTarget);
+
+            int agentPort = TcpPortProvider.GetOpenPort();
+            int aspNetCorePort = TcpPortProvider.GetOpenPort();
+
+            using (var agent = new MockZipkinCollector(Output, agentPort))
+            using (Process process = StartSample(agent.Port, arguments: null, packageVersion: string.Empty, aspNetCorePort: aspNetCorePort, startupHook: true))
+            {
+                var wh = new EventWaitHandle(false, EventResetMode.AutoReset);
+
+                process.OutputDataReceived += (sender, args) =>
```

Comment:
Could you check if ProcessHelper.Drain() can help you here. Some of the older samples contain too much boilerplate code.